AzureCalling/Controllers/AudioDeviceSelectionViewController.swift

```diff
@@ -0,0 +1,96 @@
+    var audioDeviceTableDataSource: TableViewDataSource!
+    var deviceDrawerConstraint: NSLayoutConstraint?
+
+    // MARK: IBOutlets
+
+    @IBOutlet weak var deviceDrawer: UITableView!
+
+    // MARK: UIViewController events
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        view.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.8)
+        view.isOpaque = false
+
+        let audioDeviceCell = UINib(nibName: "BottomDrawerCellView",
+                                      bundle: nil)
+        deviceDrawer.layer.cornerRadius = 8
+        deviceDrawer.register(audioDeviceCell, forCellReuseIdentifier: "BottomDrawerCellView")
+        deviceDrawer.dataSource = audioDeviceTableDataSource
+        deviceDrawer.delegate = self
+        audioDeviceTableDataSource?.deselectAllRows()
+        let currentAudioDevice = audioDeviceSelectionManager.getCurrentAudioDevice()
+        audioDeviceTableDataSource?.selectRow(title: currentAudioDevice.name)
+        deviceDrawer.reloadData()
+        hideDeviceDrawer()
+        showDeviceDrawer()
+    }
+
+    // MARK: Private Functions
+
+    private func hideDeviceDrawer() {
+        if deviceDrawerConstraint != nil {
+            self.view.removeConstraint(deviceDrawerConstraint!)
+        }
+        let hideConstraint = NSLayoutConstraint(item: deviceDrawer!,
+                attribute: .top,
+                relatedBy: .equal,
+                toItem: self.view,
+                attribute: .bottom,
+                multiplier: 1,
+                constant: 0)
+        self.view.addConstraint(hideConstraint)
+        deviceDrawerConstraint = hideConstraint
+        self.view.layoutIfNeeded()
+    }
+
+    private func showDeviceDrawer() {
+        if deviceDrawerConstraint != nil {
+            self.view.removeConstraint(deviceDrawerConstraint!)
+        }
+        let centerYConstraint = NSLayoutConstraint(item: deviceDrawer!,
+                attribute: .bottom,
+                relatedBy: .equal,
+                toItem: self.view,
+                attribute: .bottom,
+                multiplier: 1,
+                constant: 0)
+        self.view.addConstraint(centerYConstraint)
+        deviceDrawerConstraint = centerYConstraint
+        UIView.animate(withDuration: 0.25, delay: 0, options: .beginFromCurrentState, animations: {
+            self.view.layoutIfNeeded()
+        }, completion: nil)
+    }
+
+    // MARK: UITableViewDelegate
+
+    func tableView(_ tableView: UITableView,
+                   didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath) as! BottomDrawerCellView
+        let audioDeviceDataModel = AudioDeviceDataModel(image: cell.avatar.image!, name: cell.title.text!, enabled: false)
+        audioDeviceSelectionManager.switchAudioDevice(audioDeviceDataModel: audioDeviceDataModel)
```

Comment:
For selected row, we could get the selection datasource by
```
let selectedRow = indexPath.row
let selectedObject = datasource[selecedRow]
```

Could we using a closure here to pass the selectedRow to the view which holds the drawer? Like 



```
let selectedRow = indexPath.row
didSelectedData(selectedRow)
```

And in the CallViewController
```
didSelectedData =  { index in 
  let selected = AudioOutputDataType[index]
  Manager.updatedSelectedDevie(selected)
}

```